azkaban-common/src/main/java/azkaban/jobExecutor/AbstractProcessJob.java

```diff
@@ -81,6 +84,31 @@ protected void resolveProps() {
     jobProps = PropsUtils.resolveProps(jobProps);
   }
 
+  /**
+   * prints the current Job props to the Job log.
+   */
+  protected void logJobProperties() {
+    if (this.jobProps != null && 
+        this.jobProps.getBoolean(JOB_DUMP_PROPERTIES_IN_LOG, false)){
+      try {
+        Map<String,String> flattenedProps = this.jobProps.getFlattened();
+        this.info("******   Job properties   ******");
+        this.info(String.format("- Note : value is masked if property name ends with '%s'.",
+            SENSITIVE_JOB_PROP_NAME_SURFFIX ));
+        for(Map.Entry<String, String> entry : flattenedProps.entrySet()){
+          String key = entry.getKey();
+          String value = key.endsWith(SENSITIVE_JOB_PROP_NAME_SURFFIX)?
+                                      SENSITIVE_JOB_PROP_VALUE_PLACEHOLDER :
+                                      entry.getValue();
+          this.info(String.format("%s=%s",key,value));
+        }
+        this.info("****** End Job properties  ******");
+      } catch (Exception ex){
+        log.error("failed to log job properties ", ex);
```

Comment:
it goes to the job log in case there are no exceptions during the process, if for reason the process throws exception, the exception will go to Azkaban log (as I figured it doesn't make sense to try again to log to Job log when "logging to job log" is throwing exception, when that happens we just log the incident in the azkaban log and move on)